harness/tests/cli/test_auth.py

```diff
@@ -43,9 +43,263 @@ def test_auth_with_store(requests_mock: requests_mock.Mocker, user: Optional[str
+        #            |    |    |    DET_PASS env var (y="env_pass, e=None)
+        #            |    |    |     |   DET_USER_TOKEN (y="env_token", e=None)
+        #            |    |    |     |    |   token cache state (y="cache", n=empty, x="expired")
+        #            |    |    |     |    |    |   Results...
+        #            |    |    |     |    |    |    |
+        ScenarioSet("y", "y", "ne", "*", "*", "y", Check("cache")),
+        ScenarioSet("y", "y", "ne", "*", "*", "n", Login("user", "req_pass")),
+        ScenarioSet("y", "y", "ne", "*", "*", "x", Check("expired"), Login("user", "req_pass")),
+        # ---
+        # Explicit user, but password not provided.  Still no (relevant) environment settings.
+        ScenarioSet("y", "n", "ne", "*", "*", "y", Check("cache")),
+        ScenarioSet("y", "n", "ne", "*", "*", "n", Login("user", "prompt_pass")),
+        ScenarioSet("y", "n", "ne", "*", "*", "x", Check("expired"), Login("user", "prompt_pass")),
+        # ---
+        # DET_USER_TOKEN is overridden by a configured cache (the on-cluster `det user login` case).
+        ScenarioSet("*", "*", "ye", "*", "y", "y", Check("cache")),
+        # DET_USER_TOKEN can be used if user is explicit, so long as it matches.
+        ScenarioSet("y", "*", "y", "*", "y", "n", Use("env_token")),
+        # Token in env is used if cache is missing or invalid.
+        ScenarioSet("n", "*", "y", "*", "y", "n", Use("env_token")),
+        ScenarioSet("n", "*", "y", "*", "y", "x", Check("expired"), Use("env_token")),
+        # Token in env is overridden by a configured cache (the on-cluster `det user login` case).
+        ScenarioSet("n", "*", "ye", "*", "y", "y", Check("cache")),
+        # ---
+        # Explicit user and password, DET_USER set but DET_USER_TOKEN not set.  DET_PASS is ignored.
+        ScenarioSet("y", "y", "y", "*", "n", "y", Check("cache")),
+        ScenarioSet("y", "y", "y", "*", "n", "n", Login("user", "req_pass")),
+        ScenarioSet("y", "y", "y", "*", "n", "x", Check("expired"), Login("user", "req_pass")),
+        # ---
+        # Explicit user but no password, DET_USER and DET_PASS set, and DET_USER_TOKEN unset.
+        # DET_PASS still ignored since DET_USER/DET_PASS are meant to be processed as a unit, and
+        # an explicitly-requested username overrides that unit.
+        ScenarioSet("y", "n", "y", "y", "n", "y", Check("cache")),
+        ScenarioSet("y", "n", "y", "y", "n", "n", Login("user", "prompt_pass")),
+        ScenarioSet("y", "n", "y", "y", "n", "x", Check("expired"), Login("user", "prompt_pass")),
+        # ---
+        # Explicit user but no password, DET_USER set but no other env.  DET_USER is ignored.
+        ScenarioSet("y", "n", "y", "n", "n", "y", Check("cache")),
+        ScenarioSet("y", "n", "y", "n", "n", "n", Login("user", "prompt_pass")),
+        ScenarioSet("y", "n", "y", "n", "n", "x", Check("expired"), Login("user", "prompt_pass")),
```

Comment:
I feel like we should move these explicit user cases up to the top with the first set of explicit user scenarios. 
This would make it a bit easier to distinguish the purpose of each explicit user scenario since these in these cases, `DET_USER` is the same as the explicit user while in the first set of scenarios, `DET_USER` was either different or unset.
If we do leave these test cases in place, maybe mention in the test comments that env user == explicit user just so that the tests are slightly easier to understand without having to know the respective positions of the explicit user and `DET_USER` arguments in the `ScenarioSet`.